serde_v8/de.rs

```diff
@@ -661,3 +661,57 @@ fn bigint_to_f64(b: v8::Local<v8::BigInt>) -> f64 {
     .sum();
   sign * x
 }
+
+pub fn to_utf8(
+  s: v8::Local<v8::String>,
+  scope: &mut v8::HandleScope,
+) -> std::string::String {
+  if let Some(s) = to_utf8_fast(s, scope) {
+    return s;
+  }
+  to_utf8_slow(s, scope)
+}
+
+fn to_utf8_fast(
+  s: v8::Local<v8::String>,
+  scope: &mut v8::HandleScope,
+) -> Option<std::string::String> {
+  // Over-allocate by 20% to avoid checking string twice
+  let len = s.length();
+  let capacity = (s.length() as f64 * 1.2) as usize;
+  let mut buf = Vec::with_capacity(capacity);
+  let mut nchars = 0;
+  let data = buf.as_mut_ptr();
+  let length = s.write_utf8(
+    scope,
+    unsafe { std::slice::from_raw_parts_mut(data, capacity) },
+    Some(&mut nchars),
+    v8::WriteOptions::NO_NULL_TERMINATION
+      | v8::WriteOptions::REPLACE_INVALID_UTF8,
+  );
+  unsafe {
+    buf.set_len(length);
+  }
+  if nchars < len {
+    return None;
+  }
+  Some(unsafe { std::string::String::from_utf8_unchecked(buf) })
+}
+
+fn to_utf8_slow(
+  s: v8::Local<v8::String>,
+  scope: &mut v8::HandleScope,
+) -> std::string::String {
+  let capacity = s.utf8_length(scope);
+  let mut string = std::string::String::with_capacity(capacity);
+  let data = string.as_mut_ptr();
+  std::mem::forget(string);
```

Comment:
Maybe we can avoid `forget` and the `String::from_raw_parts` re-construct by just returning `string` at the end? (mutable reference does not live longer)

Reply:
That code was pretty much `v8::String::to_rust_string_lossy` untouched, I've now aligned it follow the same general design as the fast path and maybe we'll upstream these changes (or in part) to rusty_v8